### DISCUSSION TITLE
IFUS Outdoor, disable on-camera motion detection

### DISCUSSION THREAD

**Discussion by ivanobressan** [roleoroleo/yi-hack-Allwinner-v2#775] [Q&A]
Hi, i use several IFUS Outdoor camera with frigate (HA module), one of theese is pointed in a public street and i noticed when a lot of motion events occours, rstp server sends to frigate some bad data an i often remain without video.
Since i use frigate with a coral TPU to detect objets in camera, i don't need this is done by the camera itself because i think the "processor" used is not strong enough to follow the events and streaming in meanwhile.
Is there a way to turn off this on-camera detection? 
Thanks!

**Comment by roleoroleo** [comment]
It depends on the Yi fw.
With fw 12 you can probably disable motion detection.

But I recently fixed a bug in the RTSP server that could fix (maybe) your problem. I can send a beta if you want, in case give me your suffix.
Another improvement: if it's not important for your usage, you should disable the snapshot feature because is cpu/memory expensive.

**Comment by ivanobressan** [comment]
Hi, i did some tests:
it turns out if i switch cloud on, cameras are almost perfect, but if i turn cloud off, i get 5/6 rtsp server crash for every minute.
Since i hacked my cameras to non use chinese cloud, is there a way to make theese works in frigate without this """feature"""?

**Comment by roleoroleo** [comment]
Disable cloud means no chinese servers.
There is no way.
But please, test a new beta release.

**Reply by ivanobressan** [reply]
Sure! can you please give me a link to download this beta?

**Reply by roleoroleo** [reply]
What's the suffix?

**Reply by ivanobressan** [reply]
IFUS, thanks!

**Reply by roleoroleo** [reply]
I mean the model suffix.
y21ga, r30gb, ...

**Reply by ivanobressan** [reply]
Oh! Sorry!
it's H30GA

**Reply by roleoroleo** [reply]
Here it is:
[h30ga_0.3.0.tar.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/13785388/h30ga_0.3.0.tar.gz)

**Comment by nemonoid** [comment]
@ivanobressan  I have noticed the same issue, when too much of the image changes all of a sudden, the RSTP stream in my Shinobi DVR freezes for about a second.

But I don't think this is due to the CPU or RAM being overloaded.  In fact, if I connect to the YI-Hack camera with SSH and monitor the CPU/RAM usage using `top`  I can see that there is about 20% idle CPU and at least 10MB of free RAM even under intense image changes.

Even when I get these cuts of the RSTP stream, it seems that the CPU and RAM were far from being overloaded.

I will be happy to test the new RTSP version from @roleoroleo as well

**Reply by roleoroleo** [reply]
Which suffix?

**Reply by nemonoid** [reply]
Firmware Version | 0.3.0
Base Version    | 9.0.35.00_
Model Suffix    | y291ga
Serial Number | RFUS

Thank You :-)

**Comment by roleoroleo** [comment]
Here it is:
[y291ga_0.3.0.tar.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/13789471/y291ga_0.3.0.tar.gz)

**Reply by nemonoid** [reply]
Thank you.

instead of rewriting the whole SD Card I just replaced the **rRTSPServer** and **rtsp_server_yi** files, is that ok ?

**Reply by roleoroleo** [reply]
Yes.

**Comment by nemonoid** [comment]
It seems the new RTSP binary fixed the problem for me!

BEFORE: when a lot of the frame was changing all at once (e.g. a person walking beside the camera) I would get pixelation and the image would freeze for about a second.
NOW: I tried the same situations and it's not happening anymore.

For reference I'm using RTSP High (normal version, not alternative) with AAC audio, and I'm watching it on Shinobi DVR.

I will check for a few more days and I will give you a definitive answer.

**Reply by roleoroleo** [reply]
I fixed a problem about the timing of the frames.
In the old versions I assigned a frame duration for all frames (SPS and PPS included).
So the stream was losing 100ms every 2 seconds.
It's not strictly related to a scene change but it can happen when there is a big frame.

**Reply by nemonoid** [reply]
Great job :-)

**Comment by nemonoid** [comment]
After 2 days with the new binary I can confirm that I don't have the issue of "freezes" in the RTSP stream anymore. This is a major improvement!

**Reply by roleoroleo** [reply]
Good news! Thank you.
I will release a new version asap.

**Comment by ivanobressan** [comment]
Hi guys, i'm testing a lot, i can confirm a giant leap in quality, let me do some more test in infrared mode, because i noticed some issues during last night...

**Comment by nemonoid** [comment]
After about 3 weeks I can confirm that with the new RTSP binary I didn't get the "frozen image for about 1-2 seconds" problem anymore.

Tested with RTSP High (normal version, not alternative) using Shinobi and Motioneye DVRs.

@ivanobressan  any more feedback from your side ?

**Comment by roleoroleo** [comment]
This is the "big" change:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/commit/027cc477035af26745cc8d8af3411f1cc1d57f36
